mlkem/reduce.h

```diff
@@ -29,17 +30,21 @@ ENSURES(RETURN_VALUE >= -HALF_Q && RETURN_VALUE <= HALF_Q);
  * Description: Multiplication followed by Montgomery reduction
  *
  * Arguments:   - int16_t a: first factor
- *              - int16_t b: second factor
+ *                  Can be any int16_t.
+ *              - int16_t b: second factor.
+ *                  Must be signed canonical (abs value <|q|/2)
```

Comment:
What do you mean by `|q|`? Also the code checkes <= q/2, not < q/2 (as q/2 is rounded down).

Reply:
I always mean division in the rationals, and with that interpretation of q/2 the comment is correct. However, it's not obvious that it's meant this way, and one should avoid any confusion by using < (q+1)/2 explicitly.

Reply:
I will fix this in the follow-up PRs.